trainers/mlp_benchmark.py

```diff
@@ -4,185 +4,173 @@
+    X_s = np.random.normal(loc=[0., 0.], scale=[1., 1.], size=[10000, 2])
+    X_b = np.random.normal(loc=[10., 10.], scale=[1., 1.], size=[10000, 2])
+    y_s = np.zeros(10000) + 1
+    y_b = np.zeros(10000)
+    X = np.concatenate([X_s, X_b])
+    y = np.concatenate([y_s, y_b])
+    Y = np.array([y, -(y-1)]).T
+    X, Y = shuffle(X, Y)
+
+    # define config json and parameters
+    config = {'model_name': 'mlp_0321',
+              'learning_rate': 0.001,
+              'training_epochs': 10,
+              'batch_size': 100}
+    n_input = 2
+    n_hidden = [3, 3]
+    n_classes = 2
+
+    # train model
+    mlp = MLP(n_input, n_hidden, n_classes, config)
+    mlp.train(X, Y)
+    # test model
+    mlp.evaluate(X, Y)
+
+    return
 
 
 if __name__ == '__main__':
-	#test()
-	import sys
-        import time
-	from os import path
-	lib_path = path.dirname(path.dirname(path.abspath(__file__)))
-	sys.path.append(lib_path)
-	from preprocess.PreProcess import PreProcess
-	from preprocess.LearningDataAdapter import LearningDataAdapter
-	from util.resampling import binary_downsampling, binary_upsampling
-
-	print 'Loading data and preprocessing ......'
-        start_time = time.time()
-	# adapter
-	adapter = LearningDataAdapter(for_learning=True)
-	adapter.adapt_file('../data/train.csv')
-	X_num, X_cat = adapter.X_num, adapter.X_cat
-	w, y = adapter.w, adapter.y
-	# processor
-	processor = PreProcess()
-	processor.fit(X_num, X_cat, 
-	              {'imputer': '../preprocess/imputer.pkl',
-	               'scaler': '../preprocess/scaler.pkl',
-	               'encoder': '../preprocess/encoder.pkl'})
-	X = processor.transform(X_num, X_cat)
-	# resample
-	X_train, y_train = binary_upsampling(X, y)
-	Y_train = np.array([y_train, -(y_train-1)]).T
-
-        end = time.time()
-        print 'Done. Took {} seconds.'.format(end - start_time)
-	print 
-
-	print 'Training ......'
-        start = time.time()
-	# train
-	config = {'model_name': 'mlp_0321',
-			  'learning_rate': 0.001,
-			  'training_epochs': 10,
-			  'batch_size': 100}
-	mlp = MLP(X_train.shape[1], [100, 100], 2, config)
-	mlp.train(X_train, Y_train)
-        end = time.time()
-        print 'Done. Took {} seconds.'.format(end - start)
-        print
-
-	mlp.evaluate(X_train, Y_train)
-        end = time.time()
-        print 'Complete. Took total of {} seconds.'.format(end - start_time)
-        print
+    #test()
+    import sys
+    import time
+    from os import path
+    lib_path = path.dirname(path.dirname(path.abspath(__file__)))
+    sys.path.append(lib_path)
+    from preprocess.PreProcess import PreProcess, load_data
+    from util.resampling import binary_downsampling, binary_upsampling
+
+    print 'Loading data and preprocessing ......'
+    start_time = time.time()
+    process_path = {'imputer': '../preprocess/imputer.pkl', 'scaler': '../preprocess/scaler.pkl', 'encoder': '../preprocess/encoder.pkl'}
+    X, y, w = load_data('../data/train.csv', process_path, fit=False)
```

Comment:
Yes. If `fit=True`, it will fit and save .pkl files at process_path.
If `fit=False`, and there are no provided .pkl files, joblib will through Error of no such file or directory.